minidump-stackwalk/src/main.rs

```diff
@@ -16,56 +16,108 @@ use minidump_processor::{
     http_symbol_supplier, simple_symbol_supplier, MultiSymbolProvider, ProcessorOptions, Symbolizer,
 };
 
-use clap::{crate_version, App, AppSettings, Arg};
+use clap::{crate_version, App, AppSettings, Arg, ArgGroup};
 use log::error;
 use simplelog::{
     ColorChoice, ConfigBuilder, Level, LevelFilter, TermLogger, TerminalMode, WriteLogger,
 };
 
 #[cfg_attr(test, allow(dead_code))]
 fn main() {
-    let matches = App::new("minidump_stackwalk")
+    let matches = App::new("minidump-stackwalk")
         .version(crate_version!())
-        .about("Analyzes minidumps and produces a machine-readable JSON report.")
+        .about("Analyzes minidumps and produces a report (either human-readable or JSON).")
         .setting(AppSettings::NextLineHelp)
         .setting(AppSettings::ColoredHelp)
+        .setting(AppSettings::DeriveDisplayOrder)
+        .usage("minidump-stackwalk [FLAGS] [OPTIONS] <minidump> [--] [symbols-path]...")
+        .arg(
+            Arg::with_name("json")
+                .long("json")
+                .long_help("Emit a machine-readable JSON report (the default).
+
+Official schema TBD, but the output *is* stabilized. That said, sometimes values can't be computed \
```

Comment:
FYI the original implementation bug [had a quasi-schema attached](https://bug573100.bmoattachments.org/attachment.cgi?id=549163). It's assuredly out of date and not an actual JSON schema in any event.

Reply:
yeah process_state's comments are nearly a complete schema already, I just need to pull them out to a proper location and add some more notes.